.github/workflows/pr-test.yml

```diff
@@ -0,0 +1,181 @@
+name: Pull Request Testing
+
+# Trigger on pull request events
+on:
+  pull_request:
+    branches: [ main ]
```

Comment:
branch should be feature/AzCVMEmu, instead of main. Emulation support is not merged to main branch yet.